DashSync/shared/Models/Masternode/DSMasternodeList.m

```diff
@@ -634,4 +643,26 @@ - (DSSimplifiedMasternodeEntry *)masternodeForRegistrationHash:(UInt256)registra
     return self.simplifiedMasternodeListDictionaryByReversedRegistrationTransactionHash[uint256_data(registrationHash)];
 }
 
+- (void)applyWhiteList:(NSArray *)whiteList {
+    NSMutableDictionary<NSData *, DSSimplifiedMasternodeEntry *> *whiteEntries;
+    // If there's no white list entries we'll use whole list
+    if (whiteList && [whiteList count] > 0) {
+        DSAddress a;
+        whiteEntries = [NSMutableDictionary dictionary];
+        for (NSValue *address in whiteList) {
+            [address getValue:&a];
+            for (NSData *hash in self.mSimplifiedMasternodeListDictionaryByReversedRegistrationTransactionHash.allKeys) {
+                DSSimplifiedMasternodeEntry *entry = self.mSimplifiedMasternodeListDictionaryByReversedRegistrationTransactionHash[hash];
+                if (uint128_eq(entry.address.ipAddress, a.ipAddress) &&
+                    entry.address.port == a.port) {
+                    whiteEntries[hash] = entry;
+                }
+            }
+        }
+    } else {
+        whiteEntries = self.mSimplifiedMasternodeListDictionaryByReversedRegistrationTransactionHash;
+    }
+    self.mSimplifiedMasternodeWhiteListDictionaryByReversedRegistrationTransactionHash = whiteEntries;
```

Comment:
This can't be modified otherwise the system will not work any more. Right now (before the PR), I think what we do is just take a node from the last masternode list. What we need to do is that when we try to get a node for Dapi, we just do an intersection of nodes on the masternode list, and nodes in the whitelist, and just take a random one in the intersection.